### PR TITLE
Fix JWT retry logic

### PR DESCRIPTION
**Pull request by sujaygarlanka**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sujaygarlanka has signed the CLA. Thanks for the pull request!

File: lib/token-manager.js [hunk 1]
```diff
@@ -345,37 +349,92 @@ TokenManager.prototype = {
...
+				if (typeof retryTimeout !== 'number') {
+					if (retryTimeout instanceof Error) {
+						error = retryTimeout;
 					}
-
-					return this.getTokens(params, options);
+					throw error;
 				}
+			} else if (error.response.headers.hasOwnProperty('retry-after')) {
+				retryTimeout = error.response.headers['retry-after'] * 1000;
```
Comment (PJSimon) [inline lib/token-manager.js]: Should you add the retry-after check here, as well?
https://github.com/box/box-node-sdk/blob/master/lib/api-request.js#L321

File: tests/lib/token-manager-test.js [hunk 2]
```diff
@@ -630,7 +629,181 @@ describe('token-manager', function() {
...
+		}).timeout(5000);
+
+		it('should retry with new jti claim when server rejects because of rate limiting', function() {
+
+			var firstTokenParams = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+				assertion: TEST_WEB_TOKEN
+			};
+
+			var newJWT = 'jhdhioeiu4yo34875twudgshdgr';
```
Comment (PJSimon) [inline tests/lib/token-manager-test.js]: How was this generated?  It should be a freshly-generated key from a non-Box account...

File: tests/lib/token-manager-test.js [hunk 3]
```diff
@@ -630,7 +629,181 @@ describe('token-manager', function() {
...
+			var secondTokenParams = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+				assertion: newJWT
+			};
+
+			var newJWT2 = 'jhighreiu4yo34875tfreggshdgr';
```
Comment (PJSimon) [inline tests/lib/token-manager-test.js]: How was this generated?  It should be a freshly-generated key from a non-Box account...

File: tests/lib/token-manager-test.js [hunk 4]
```diff
@@ -630,7 +629,181 @@ describe('token-manager', function() {
...
+					},
+					headers: {
+						'retry-after': 2,
+						date: serverDate2
+					}
+				}
+			};
+
+			var tokenInfo = {
+				accessToken: 'lsdjhgo87w3h4tbd87fg54'
```
Comment (PJSimon) [inline tests/lib/token-manager-test.js]: How was this generated? It should be a freshly-generated key from a non-Box account...

Comment (sujaygarlanka) [reply tests/lib/token-manager-test.js]: This is taken from other tests in the file.

File: tests/lib/token-manager-test.js (hunk 2 above)
Comment (sujaygarlanka) [reply tests/lib/token-manager-test.js]: This is taken from other tests in the file.

File: tests/lib/token-manager-test.js (hunk 3 above)
Comment (sujaygarlanka) [reply tests/lib/token-manager-test.js]: I just changed up some characters in the newJWT parameter.

**Pull request merged**